R/check_filename.R

```diff
@@ -1,4 +1,38 @@
-#' Check the style of file names
+#' Check the style of file and folder names
+#'
+#' A consistent naming schema avoids problems when working together.
+#' Especially when working with different OS.
+#' Some OS (e.g. Windows) are case-insensitive whereas others (e.g. Linux) are
+#' case-sensitive.
+#' Note that the the checklist GitHub Actions will test your code on Linux,
+#' Windows and MacOS.
+#'
+#' The sections below describe the default rules.
+#' We allow several exceptions when the community standard is different.
+#' E.g. a package stores the function scripts in the `R` folder, while our
+#' standard we enforces lower case folder names.
+#' Use the the community standard, even if it does conform with the checklist
+#' rules.
+#' Most likely checklist will have an exception for the name.
+#' If not, you can file an [issue](https://github.com/inbo/checklist/issues) and
+#' motivate why you think we should add an exception.
+#'
+#' @section Rules for folder names:
+#' - Folder names should only contain lower case letters, numbers and
+#'   underscore (`_`).
+#' - They can start with a single dot (`.`).
+#' @section Rules for file names:
+#' - Base names should only contain lower case letters, numbers and
+#'   underscore (`_`).
+#' - File extensions should only contains lower case letters and numbers.
+#'   Exceptions: file extensions related to `R` must have an upper case `R` (
+#'   `.R`, `.Rmd`, `.Rd`, `.Rnw`, `.Rproj`).
+#'
+#' @section Rules for graphical file names:
```

Comment:
Suggestion: if the previous section would become longer, you might want to start the previous section with mentioning that it only applies to file names of non graphical files (to avoid users overlook this section as the previous one seems to cover all file names).  On the other hand: if error messages are clear, users will find out anyway.  ;-)